src/Components/Server/src/Circuits/CircuitIdFactory.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Components.Server.Circuits
+{
+    // This is a singleton instance
+    // Generates strong criptographic ids for circuits that are protected with authenticated encryption.
+    internal class CircuitIdFactory
+    {
+        // Follow up with blowdart/levi. Create should be good enough, but not sure if we need
+        // to do anything additional for FIPS compliance.
+        private readonly RandomNumberGenerator _generator = RandomNumberGenerator.Create();
```

Comment:
I've searched through the codebase and this is how we do it in all places, so this must be the correct way to create it. The only reason I made this comment in the first place is because for some crypto algorithms there are special ways to create them in order to achieve FIPS compliance. This is not the case with the PRNG I believe

Reply:
PRNGs aren't part of FIPS, so you're ok here